modules/javafx.controls/src/test/java/test/javafx/scene/control/skin/LabelSkinTest.java

```diff
@@ -2091,6 +2093,47 @@ public void testEscapedMnemonicSymbolIsNotProcessedWhenParsingIsDisabled() {
         assertEquals("foo __bar", LabelSkinBaseShim.getText(label).getText());
     }
 
+    /*********************************************************************
+     *
+     * Tests for bug reports                                             *
+     *
+     ********************************************************************/
+
+    //Test for JDK-8172849
+    @Test
+    public void testBaselineAlignmentWhenTextIsSetAndGraphicIsSet() {
+        Label label1 = new Label("Label1");
+        LabelSkinMock label1Skin = new LabelSkinMock(label1);
+        label1.setSkin(label1Skin);
+
+        Label label2 = new Label("Label2");
+        LabelSkinMock label2Skin = new LabelSkinMock(label2);
+        label2.setSkin(label2Skin);
+
+        Rectangle r = new Rectangle(15, 5);
+        label2.setGraphic(r);
+
+        Toolkit tk = Toolkit.getToolkit();
+        HBox hBox = new HBox(label1, label2);
+        hBox.setAlignment(Pos.BASELINE_LEFT);
+        StageLoader sl = new StageLoader(hBox);
+        tk.firePulse();
+
+        double label2Height = label2.getHeight();
+        assertEquals(label1.getBaselineOffset(), label2.getBaselineOffset(), 0);
+        r.setHeight(50);
+        tk.firePulse();
+        assertEquals(label1.getBaselineOffset(), (label2.getBaselineOffset() - (r.getHeight()-label2Height)/2), 0);
+        sl.dispose();
+    }
+
+
+    /********************************************************************************
+     *                                                                              *
+     *                             Helper classes                                   *
+     *                                                                              *
+     *******************************************************************************/
+
```

Comment:
Minor: I would have used a javadoc explaining the purpose of this class instead of generic "Helper classes".

Reply:
Same comment as above. In addition to that, yes, updating the comment with actual purpose of the class would be better. I will update that once we decide if all the other comment blocks also should be changed or not.
Since there was no comment for this class, I added the comment so that it gets separated from "Tests for bug reports" tests.

Reply:
Updated the comment with purpose of the class instead of generic comment.